src/index.ts

```diff
@@ -0,0 +1,95 @@
+import { Octokit } from "@octokit/core";
+import { VERSION } from "./version";
+import { Endpoints } from "@octokit/types";
+
+const OCTOKIT_UNIQUE_ISSUE_ID_PREFIX = "octokit-unique-issue id=";
+
+type CreateOrUpdateUniqueIssueOptionsT =
+  Endpoints["POST /repos/{owner}/{repo}/issues"]["parameters"] & {
+    identifier: string;
+    update_previous?: boolean;
+  };
+
+type CreateOrUpdateUniqueIssueResponseT =
+  | Endpoints["POST /repos/{owner}/{repo}/issues"]["response"]
+  | Endpoints["PATCH /repos/{owner}/{repo}/issues/{issue_number}"]["response"];
+
+export function uniqueIssue(octokit: Octokit) {
+  return {
+    async createOrUpdateUniqueIssue(
+      options: CreateOrUpdateUniqueIssueOptionsT
+    ): Promise<CreateOrUpdateUniqueIssueResponseT> {
+      const {
+        owner,
+        repo,
+        identifier,
+        update_previous = true,
+        body,
+        ...rest
+      } = options;
+
+      if (!identifier) {
+        throw Object.assign(new Error("identifier is required."), {
+          name: "MissingIdentifierError",
+        });
+      }
+
+      const term = `${OCTOKIT_UNIQUE_ISSUE_ID_PREFIX}"${identifier}"`;
+
+      // MDAST compatible comment marker; see https://github.com/syntax-tree/mdast-comment-marker
+      const commentMarker = `<!-- ${term} -->`;
+
+      const {
+        data: { items, total_count },
+      } = await octokit.request("GET /search/issues", {
+        q: `"${term}" is:issue is:open repo:${owner}/${repo}`,
+      });
+
+      if (total_count === 1 && update_previous) {
+        return await octokit.request(
+          "PATCH /repos/{owner}/{repo}/issues/{issue_number}",
+          {
+            owner,
+            repo,
+            issue_number: items[0].number,
+            ...(body !== undefined
+              ? { body: body + `\n${commentMarker}` }
```

Comment:
would it make sense to add two line breaks to have an empty line above the marker?

```suggestion
              ? { body: body + `\n\n${commentMarker}` }
```

Reply:
We talked about this and I think this makes sense! 👍🏻